student/src/main/java/tds/student/sql/repository/remote/impl/RemoteExamSegmentWrapperRepository.java

```diff
@@ -0,0 +1,86 @@
+package tds.student.sql.repository.remote.impl;
+
+import TDS.Shared.Exceptions.ReturnStatusException;
+import com.google.common.base.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.util.List;
+import java.util.UUID;
+
+import tds.common.Response;
+import tds.exam.ExamAssessmentMetadata;
+import tds.exam.wrapper.ExamSegmentWrapper;
+import tds.student.sql.repository.remote.ExamSegmentWrapperRepository;
+
+@Repository
+public class RemoteExamSegmentWrapperRepository implements ExamSegmentWrapperRepository {
```

Comment:
This is how you get the full amount of data for an exam page and items that are required by teh Student application.

It's PageGroup consists of data from our ExamSegment, ExamPage, and ExamItems